build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/RestCompatTestTransformTask.java

```diff
@@ -367,12 +413,38 @@ public void transform() throws IOException {
         // clean the output directory to ensure no stale files persist
         fileSystemOperations.delete(d -> d.delete(outputDirectory));
 
+        Map<File, String> skippedFilesWithReason = new HashMap<>();
+        skippedTestByFilePatternTransformations.forEach((filePattern, reason) -> {
+            //resolve file pattern to concrete files
+            for (File file : getTestFiles().matching(filePattern).getFiles()) {
+                skippedFilesWithReason.put(file, reason);
+            }
+        });
+
         RestTestTransformer transformer = new RestTestTransformer();
         // TODO: instead of flattening the FileTree here leverage FileTree.visit() so we can preserve folder hierarchy in a more robust way
         for (File file : getTestFiles().getFiles()) {
             YAMLParser yamlParser = YAML_FACTORY.createParser(file);
             List<ObjectNode> tests = READER.<ObjectNode>readValues(yamlParser).readAll();
-            List<ObjectNode> transformRestTests = transformer.transformRestTests(new LinkedList<>(tests), transformations);
+            List<ObjectNode> transformRestTests;
```

Comment:
I assume we have to do all this stuff _here_ because we can exclude based on full test name, which includes the file path, which I guess is only known here where we are actually iterating the files? We otherwise don't pass the file path down through the transforms?

Reply:
Correct. The Transformations themselves don't know anything about the 3/part/convention or file names, just the Java objects of the de-serialized YAML tests.